packages/kbn-unified-field-list/src/containers/unified_field_list_sidebar/field_list_sidebar.tsx

```diff
@@ -432,27 +426,26 @@ export const UnifiedFieldListSidebar = memo(UnifiedFieldListSidebarComponent);
 // eslint-disable-next-line import/no-default-export
 export default UnifiedFieldListSidebar;
 
-function calculateMultiFields(
+export function getMultiFieldsByParent(
```

Comment:
With this implementation the initial render of the list would be faster but each popover opening would be slower as it iterates again through the list. It's better not to refactor this logic in this PR.

I think that we should still keep the map of multi fields like before but build it only once and in a "lazy" way.

Reply:
agreed, let's undo this to reduce the scope 👍